source/interact-data/crud.txt

```diff
@@ -677,6 +677,156 @@ When you globally set ``join_contexts`` to ``true``, you can use the
+were made previously by calling the ``Model#previous_changes`` method,
+as shown in the following code:
+
+.. literalinclude:: /includes/interact-data/crud.rb
+   :language: ruby
+   :start-after: start-dirty-tracking-prev
+   :end-before: end-dirty-tracking-prev
+
+Update Container Fields
+-----------------------
+
+{+odm+} currently has an issue that prevents changes to attributes of
+container types, such as ``Set`` or ``Array``, from saving to MongoDB.
+You must assign all fields, including container types, for their values
+to save to MongoDB.
+
+For example, adding an item to a ``Set`` instance as shown in the
+following code *does not* persist changes to MongoDB:
+
+.. code-block:: ruby
+
+   person = Person.new
+   person.interests
+   # => #<Set: {}>
+
+   person.interests << 'Hiking'
+   # => #<Set: {"Hiking"}>
+   person.interests
+   # => #<Set: {}> # Change does not take effect
+
+To persist this change, you must modify the field value *outside* of the
+model and assign it back to the model as shown in the following code:
+
+.. literalinclude:: /includes/interact-data/crud.rb
+   :language: ruby
+   :start-after: start-container-save
+   :end-before: end-container-save
+
+Read-only Documents
+-------------------
+
+You can mark documents as read-only in the following ways, depending on
+the value of the ``Mongoid.legacy_readonly`` feature flag:
+
+- If this flag is turned *off*, you can mark a document as
+  read-only by calling the ``readonly!`` method on that document. The
+  resulting read-only document raises a ``ReadonlyDocument`` error if
+  you attempt to perform any persistence operation, including, but not
+  limited to, saving, updating, deleting, and destroying. Note that
+  reloading *does not* reset the read-only state.
+  
+  .. code-block:: ruby
+
+     person = Person.first
+     person.readonly? # => false
+     person.readonly! # Sets the document as read-only
+     person.readonly? # => true
+     person.name = "Larissa Shay" # Changes the document
+     person.save # => raises ReadonlyDocument error
+     person.reload.readonly? # => true
+
+- If this flag is turned ``on``, you can mark a document as read-only
+  when that document is projected, such as by using ``only`` or
+  ``without``. The resulting read-only document is not deletable or
+  destroyable, as {+odm+} raises a ``ReadonlyDocument`` error, but it is
+  saveable and updatable. The read-only status *is reset* if you reload the
+  document.
+
+  .. code-block:: ruby
+
+     person = Person.only(:name).first
+     person.readonly? # => true
+     person.destroy # => raises ReadonlyDocument error
+     person.reload.readonly? # => false
+
+.. tip:: Projection
+
+   To learn more about projections, see the
+   :ref:`mongoid-data-projection` section of the Modify Query
+   Results guide.
+
+Override readonly?
+~~~~~~~~~~~~~~~~~~
+
+You can also make a document read-only by overriding the ``readonly?``
+method, as shown in the following code:
```

Comment:
Since the previous section says: "You can make documents readonly in the following ways..." I think this can be a part of that section instead of having a dedicated header
```suggestion
You can also make a document read-only by overriding the ``readonly?``
method, as shown in the following code:
```